common/nextpnr.h

```diff
@@ -833,6 +833,7 @@ struct BaseCtx
 
     // Top-level ports
     std::unordered_map<IdString, PortInfo> ports;
+    std::unordered_map<IdString, CellInfo *> port_cells;
```

Comment:
can you add a brief note about `port_cells` to `docs/netlist.md`?

Reply:
Done.  

There is a broader question about separating the root level cell namespace and the root level port namespace.  In Verilog I believe it is legal to have root level ports and root level cells with the same name, but this causes a failure in nextpnr.  I believe the "correct" long term answer is to have a separate namespace for root level ports and cell names, but I don't think this needs to be fixed right now.

Reply:
I think it could also make sense to use a new type (eg TopLevelPortInfo) other than PortInfo here that could store things like attributes and also partition pin information for partially reconfigurable designs. But this is definitely not for this PR but another day.